src/GraphQL/Validation/Rules.Custom/ComplexityValidationRule.cs

```diff
@@ -35,16 +36,49 @@ public ComplexityValidationRule(ComplexityConfiguration complexityConfiguration,
 
     /// <inheritdoc/>
     public ValueTask<INodeVisitor?> ValidateAsync(ValidationContext context)
+        => new(new PostponeComplexityValidationVisitor(_complexityConfiguration, _complexityAnalyzer));
+
+    // https://github.com/graphql-dotnet/graphql-dotnet/issues/3527
+    private sealed class PostponeComplexityValidationVisitor : INodeVisitor
```

Comment:
@Shane32 I think to revert all files except this one and `recursive_fragment_should_not_end_in_eternal_loop1`.

Reply:
Nope. KISS pattern made a job.